packages/@aws-cdk/aws-servicediscovery/lib/service.ts

```diff
@@ -0,0 +1,448 @@
+  name?: string;
+
+  /**
+   * A description of the service.
+   *
+   * @default none
+   */
+  description?: string;
+
+  /**
+   * Structure containing failure threshold for a custom health checker.
+   * Only one of healthCheckConfig or healthCheckCustomConfig can be specified.
+   * See: https://docs.aws.amazon.com/cloud-map/latest/api/API_HealthCheckCustomConfig.html
+   *
+   * @default none
+   */
+  healthCheckCustomConfig?: HealthCheckCustomConfig;
+}
+
+/**
+ * Service props needed to create a service in a given namespace. Used by createService() for PrivateDnsNamespace and
+ * PublicDnsNamespace
+ */
+export interface DnsServiceProps extends BaseServiceProps {
+  /**
+   * The DNS type of the record that you want AWS Cloud Map to create. Supported record types
+   * include A, AAAA, A and AAAA (A_AAAA), CNAME, and SRV.
+   *
+   * @default A
+   */
+  dnsRecordType?: DnsRecordType;
+
+  /**
+   * The amount of time, in seconds, that you want DNS resolvers to cache the settings for this
+   * record.
+   *
+   * @default 60
+   */
+  dnsTtlSec?: number;
+
+  /**
+   * A complex type that contains settings for an optional health check.
+   * If you specify settings for a health check, AWS Cloud Map associates the health check with the
+   * records that you specify in DnsConfig.
+   * Public DNS namespaces only. Only one of healthCheckConfig or healthCheckCustomConfig can be specified.
+   *
+   * @default none
+   */
+  healthCheckConfig?: HealthCheckConfig;
+
+  /**
+   * The routing policy that you want to apply to all DNS records that AWS Cloud Map creates when you
+   * register an instance and specify this service.
+   *
+   * @default WEIGHTED for CNAME records, MULTIVALUE otherwise
+   */
+  routingPolicy?: RoutingPolicy;
+
+  /**
+   * Whether or not this service will have an Elastic LoadBalancer registered to it as an AliasTargetInstance
+   *
+   * @default false
+   */
+
+  loadBalancer?: boolean;
+}
+
+export interface ServiceProps extends DnsServiceProps {
+  /**
+   * The ID of the namespace that you want to use for DNS configuration.
+   */
+  namespace: INamespace;
+}
+
+/**
+ * Define a CloudMap Service
+ */
+export class Service extends cdk.Construct implements IService {
+  /**
+   * A name for the Cloudmap Service.
+   */
+  public readonly serviceName: string;
+
+  /**
+   *  The namespace for the Cloudmap Service.
+   */
+  public readonly namespace: INamespace;
+
+  /**
+   * The ID of the namespace that you want to use for DNS configuration.
+   */
+  public readonly serviceId: string;
+
+  /**
+   * The Arn of the namespace that you want to use for DNS configuration.
+   */
+  public readonly serviceArn: string;
+
+  /**
+   * The DnsRecordType used by the service
+   */
+  public readonly dnsRecordType: DnsRecordType;
+
+  /**
+   * The Routing Policy used by the service
+   */
+  public readonly routingPolicy: RoutingPolicy;
+
+  // FIXME make this only called through #createService on namespace classes?
```

Comment:
I don't think that's needed. It's okay if people want to instantiate directly. Should be the same